commcare_connect/events/models.py

```diff
@@ -0,0 +1,144 @@
+from abc import ABCMeta, abstractproperty
+from dataclasses import dataclass
+from datetime import datetime
+
+from django.db import models
+from django.utils.translation import gettext as _
+
+from commcare_connect.users.models import User
+
+from . import types
+
+
+class Event(models.Model):
+    from commcare_connect.opportunity.models import Opportunity
+
+    # this allows referring to event types in this style: Event.Type.INVITE_SENT
+    Type = types
+
+    date_created = models.DateTimeField(auto_now_add=True, db_index=True)
```

Comment:
If these events are processed asynchronously, I don't think we can use `auto_now_add` especially since there is data from the phone that could be hours or days delayed.

Reply:
Hmm, that's a good point. Though, this field is [getting overridden](https://github.com/dimagi/commcare-connect/pull/349/files) when the events are created. I agree it's better to remove the auto_now_add and let the callers worry about setting it.